internal/k8s/statusaddress.go

```diff
@@ -234,8 +236,14 @@ func (s *ServiceStatusLoadBalancerWatcher) OnDelete(obj any) {
 	})
 }
 
+func (s *ServiceStatusLoadBalancerWatcher) OnElectedLeader() {
+	s.leader.Store(true)
+}
+
 func (s *ServiceStatusLoadBalancerWatcher) notify(lbstatus core_v1.LoadBalancerStatus) {
-	s.LBStatus <- lbstatus
+	if s.leader.Load() {
```

Comment:
Is it possible to lose the envoy service event in this way?

Reply:
True.  Currently the update logic relies on Added events alone. It works for other resource types as they will be processed by all Contour instances but I think new approach is needed with the load balancer status updates.

Reply:
Maybe we can simply handle it. Notify the latest envoy service when it becomes Leader.

Reply:
I've changed the approach to start the service watcher only after the instance becomes the leader.